### DISCUSSION TITLE
Create database migration method like "createIfNotExists"

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#47874] [Ideas]
Hi :wave:

I've recently found myself migrating over the migration up method from:

```php
/**
 * Run the migrations.
 */
public function up(): void
{
    Schema::create('notifications', function (Blueprint $table) {
        $table->uuid('id')->primary();
        $table->string('type');
        $table->morphs('notifiable');
        $table->text('data');
        $table->timestamp('read_at')->nullable();
        $table->timestamps();
    });
}
```

To...

```php
/**
 * Run the migrations.
 */
public function up(): void
{
    if (!Schema::hasTable('notifications')) {
        Schema::create('notifications', function (Blueprint $table) {
            $table->uuid('id')->primary();
            $table->string('type');
            $table->morphs('notifiable');
            $table->text('data');
            $table->timestamp('read_at')->nullable();
            $table->timestamps();
        });
    }
}
```

I think it would be great to have a feature such as:

```php
/**
 * Run the migrations.
 */
public function up(): void
{
    Schema::createIfNotExists('table', function (Blueprint $table) {
        // ...
    });
}
```

**Comment by TheBlckbird** [comment]
But why do you need this?

**Reply by sts-ryan-holton** [reply]
> great to have a feature

**Reply by TheBlckbird** [reply]
@sts-ryan-holton But what are the use cases for this?

**Reply by sts-ryan-holton** [reply]
In some edge cases, I've found not all packages include a `down` method for reverting migration changes, equally, for robustness I've found myself wrapping, like the example I've provided my migration in the `hasTable` method a lot, I feel that this feature should check behind the scenes if the table exists, but, I understand to provides backwards compatible support it's easier to introduce a new method such as `createIfNotExists`, which, behind the scenes can check the table existence first.

**Reply by TheBlckbird** [reply]
that makes sense
